pkg/operator/operator.go

```diff
@@ -202,40 +203,53 @@ func (c *csiDriverOperator) Run(workers int, stopCh <-chan struct{}) {
 	<-stopCh
 }
 
-func (c *csiDriverOperator) sync() error {
-	instance, err := c.client.GetOperatorInstance()
-	if err != nil {
-		if errors.IsNotFound(err) {
-			klog.Warningf("Operator instance not found: %v", err)
-			return nil
-		}
-		return err
-	}
-
+func (c *csiDriverOperator) handleDeletion(instance *v1alpha1.AWSEBSDriver) (bool, error) {
 	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
 		// CR is NOT being deleted, so make sure it has the finalizer
 		if addFinalizer(instance, operatorFinalizer) {
```

Comment:
Finalizer should not be handled in a function that handles deletion. Please move it somewhere else.

Reply:
I split this logic into a func (to update the finalizer) and a small block to delete the operand if needed. I think it's better now.